### PR TITLE
Adding MDPWS Compression Option (issue#97)

### PR DESCRIPTION
**Pull request by JavierEspina**
David, please have a look at these changes. Hopefully the content is good, well-placed, and I did not mess up in my first real editorial use of this toolchain...

### CODE REVIEW

File: SDPi_Supplement/asciidoc/volume2/mdpws-compression/tf2-ch-a-mdpws-compression.adoc [hunk 1]
```diff
@@ -0,0 +1,56 @@
+==== MDPWS Compression Option
+
+<<ieee_11073_20702_2016>> (MDPWS) defines EXI as the designated means to compress XML data. However, EXI is lagging in adoption. Therefore, instead of using EXI compression, it is recommended to use HTTP compression with gzip (RFC 1952) as the minimum agreement between clients and servers. One issue with that is the missing official support for HTTP compression in HTTP request messages. Request headers are allowed to include Content-Encoding, but those might not be accepted by HTTP servers.
+
+.R0005
+[sdpi_requirement#r0005,sdpi_req_level=should]
+****
+HTTP clients should request gzip compression by using Accept-Encoding with "gzip".
+****
+===== Example of compression request:
```
Comment (d-gregorczyk) [inline SDPi_Supplement/asciidoc/volume2/mdpws-compression/tf2-ch-a-mdpws-compression.adoc:10]: I like the idea of making examples collapsible - at least when being part of a requirement. Would you mind if we agree on moving the example into the requirement box, removing the title and making the examples a listing?

```
 .R0005
[sdpi_requirement#r1005,sdpi_req_level=should]
****
HTTP clients should request gzip compression by using Accept-Encoding with "gzip".

.Examples
[%collapsible]
====
.Request header to request compression
[source]
----
GET /path/to/resource HTTP/1.1

Host: www.example.com

Accept-Encoding: gzip
----
[...]
====
****
```

File: SDPi_Supplement/asciidoc/volume2/mdpws-compression/tf2-ch-a-mdpws-compression.adoc [hunk 2]
```diff
@@ -0,0 +1,56 @@
+==== MDPWS Compression Option
+
+<<ieee_11073_20702_2016>> (MDPWS) defines EXI as the designated means to compress XML data. However, EXI is lagging in adoption. Therefore, instead of using EXI compression, it is recommended to use HTTP compression with gzip (RFC 1952) as the minimum agreement between clients and servers. One issue with that is the missing official support for HTTP compression in HTTP request messages. Request headers are allowed to include Content-Encoding, but those might not be accepted by HTTP servers.
+
+.R0005
+[sdpi_requirement#r0005,sdpi_req_level=should]
+****
+HTTP clients should request gzip compression by using Accept-Encoding with "gzip".
```
Comment (d-gregorczyk) [inline SDPi_Supplement/asciidoc/volume2/mdpws-compression/tf2-ch-a-mdpws-compression.adoc:8]:
```suggestion
In its role as an HTTP client, a <<actor_somds_participant>> should request gzip compression by using Accept-Encoding with "gzip".
```

File: SDPi_Supplement/asciidoc/volume2/mdpws-compression/tf2-ch-a-mdpws-compression.adoc [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+Content-Type: text/xml; charset=UTF-8
+
+Content-Encoding: gzip
+|===
+The HTTP server decided to encode the response with the gzip compression. Note that servers are not required to actually compress (e.g. due to load conditions or unknown algorithms they are entitled to answer with identity encoding).
+
+====
+
+Unluckily, SDC makes heavy use of request payloads when delivering notifications. Hence, it is further recommended to allow clients to send compressed WS-Eventing Notification requests if a Subscribe request already included an accepted encoding.
```
Comment (d-gregorczyk) [inline SDPi_Supplement/asciidoc/volume2/mdpws-compression/tf2-ch-a-mdpws-compression.adoc:49]:
```suggestion
Unluckily, SDC makes heavy use of request payloads when delivering notifications. Hence, it is further recommended to allow for <<actor_somds_provider>>s to send compressed WS-Eventing Notification requests if a Subscribe request already included an accepted encoding.
```

File: SDPi_Supplement/asciidoc/volume2/mdpws-compression/tf2-ch-a-mdpws-compression.adoc [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+.R0006
+[sdpi_requirement#r0006,sdpi_req_level=may]
+****
+If an HTTP client includes an Accept-Encoding header field in an HTTP header in a WS-Eventing (W3C WS-Eventing) Subscribe request, then the Event Source may transmit Notifications related to that subscription encoded with the encoding that was defined in the Subscribe request.
```
Comment (d-gregorczyk) [inline SDPi_Supplement/asciidoc/volume2/mdpws-compression/tf2-ch-a-mdpws-compression.adoc:54]:
```suggestion
If a <<actor_somds_consumer>> includes an Accept-Encoding header field in an HTTP header in a WS-Eventing Subscribe request, the Event Source may transmit Notifications related to that subscription encoded with the encoding that was defined in the Subscribe request.
```

**Comment by JavierEspina** [comment]
@d-gregorczyk , I've accepted and fixed everything (I think) and requested your review again. What happens after that once you're happy? IOW, who does the pull into Master, closes this PR and updates that other file with changes that Todd mentioned in the issue

**Comment by d-gregorczyk** [comment]
I just fixed some minor Asciidoc markup things - other than that it looks very good to me. However, regarding the note Todd left in the issue description: should we create an additional requirement to state that if gzip compression requested, a participant shall accept the encoding?

**Comment by JavierEspina** [comment]
> I just fixed some minor Asciidoc markup things - other than that it looks very good to me. However, regarding the note Todd left in the issue description: should we create an additional requirement to state that if gzip compression requested, a participant shall accept the encoding?

I agree with the change in the last commit you made.
Regarding Todd's note: where is it? I can't find it!

**Comment by d-gregorczyk** [comment]
From the issue description:

> From the SDPi 1.0 Workshop: Supporting HTTP compression is mandatory, everything (e.g. EXI) else is not. David will put in the MDPWS appendix and Todd will provide it in SDPi-P in transport agnostic language.

Not sure if we meant to make HTTP compression mandatory, but only to the extent it is required by HTTP itself... which is also fine for me. And in that case we can merge right away.

**Comment by ToddCooper** [comment]
Thank you for the connection, David.  When I said "provide it in SDPi-P in
transport agnostic language", my reference was most probably to the profile
option (in Word supplement section 10.2.3) for "Compact Representation
Option".  This was an option we proposed because not all systems
implementing SDPi-P would support compression.  So we wanted it as an
"option".  From that SDPi-P section, we would then reference the TF-2
compression - related sections and include any requirements, including HTTP
compression as mandatory.

FWIW - The section notes from the Word draft include:

[Editor’s Note:

1.     Compression enablement

2.     NOTE:  This applies to a compressed XML representation … right?

3.     And/or does it apply to compression of streams / waveforms, specific
bulky data?

]


~>Todd 탇

*Soli Deo gloria*

M:  +1 858.442.9200
S:  ToddCooperAFC

**Comment by d-gregorczyk** [comment]
Ok, that makes sense, hence we don't need any further changes in this branch.

**Review by d-gregorczyk — APPROVED** [review]

**Comment by JavierEspina** [comment]
We had some trouble understanding each other above... But the bottom line is that I think we're fine with the closure of this PR and related issue. We can do / trigger the clean-up of the TOI page, SmartSheet row and [sdpi-supplement-issues.adoc](https://github.com/IHE/sdpi-fhir/blob/master/SDPi_Supplement/asciidoc/sdpi-supplement-issues.adoc) this Friday at the SDPi call. And we could always open a new issue later IF any DEV Domain or PC reviewer requests for more "mandatoryness" than the current recommendations.

**Pull request merged**